v3/src/hooks/use-document-context.ts

```diff
@@ -1,7 +1,7 @@
 import { createContext, useContext } from "react"
-import { IDocumentContentModel } from "../models/document/document-content"
+import { IDocumentModel } from "../models/document/document"
 
-export const DocumentContext = createContext<IDocumentContentModel | undefined>(undefined)
+export const DocumentContext = createContext<IDocumentModel | undefined>(undefined)
```

Comment:
In order to make updating the web view's url undoable, the `IDocumentModel` must be passed into `OptionsShelfButton`, so there are now two places where we need to prop drill it.

I think it's quite confusing that `useDocumentContext` provides the content rather than the document itself. This leads to situations where variables called `document` sometimes refer to `IDocumentModel` and sometimes to `IDocumentContentModel`. What do you think about changing `useDocumentContext` to `useDocumentContent` and naming variables that contain the content `documentContent`?